docs/kibana-custom-certs.md

```diff
@@ -0,0 +1,56 @@
+# Kibana Route With Custom Certificates
+Starting with the 4.0 release of cluster logging, administrators must create their own route if they
+need to utilize custom certificates.  Following are instructions for creating a re-encrypt route for Kibana:
+
+1. Extract the CA cert from the Elasticsearch secret.  This secret is created by the `cluster-logging-operator` to generate certs for the cluster logging components.  It is required to properly create the re-encrypt route
+```
+oc extract secret/elasticsearch --to=. --keys=admin-ca
+```
+2. Follow the instructions to create a [re-encrypt route](https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html#secured-routes) using the contents of the file `admin-ca` from the previous step for the destination ca certificate.
+
+```
+apiVersion: "v1"
+kind: "Template"
+metadata:
+  name: kibana-custom-route-template
+objects:
+- apiVersion: "v1"
+  kind: "Route"
+  metadata:
+    name: kibana-custom
+    namespace: openshift-logging
+  spec:
+    tls:
+      key: |
+          ${KEY}
+      certificate: |
+          ${CERT}
+      caCertificate: |
+          ${CA_CERT}
+      destinationCACertificate: |
+          ${DEST_CA_CERT}
+      termination: reencrypt
+      insecureEdgeTerminationPolicy: Redirect
+    to:
+      kind: Service
+      name: ${SERVICE_NAME}
+parameters:
```

Comment:
I'm thinking, how to fail fast, in case the users screw things up.
I'd mark all these fields as `required`. Then, if eg `cacert` doesn't exist, `CA_CERT="$(cat cacert)` will be empty string and `oc process` will fail.

Reply:
updated